src/actions/index.js

```diff
@@ -53,14 +54,24 @@ export function checkRoutesData() {
         return;
       }
 
-      const routes = routesData.map((r) => {
+      if (!state.currentRoute && state.segmentRange) {
```

Comment:
Does this unnecessarily call the API when you click on a route inside the time range?

Reply:
Also, a single line comment explaining this would be a good idea!

Reply:
Sorry for overlooking this! I've added a comment and fixed it.